### DISCUSSION TITLE
Images and fabricJS

### DISCUSSION THREAD

**Discussion by jeezs** [fabricjs/fabric.js#8651] [Q&A] [answered]
why does the following code : 
```    
var canvas = new fabric.Canvas('c');
    // create a rectangle object
    fabric.Image.fromURL('https://picsum.photos/200/300', function(img) {
        canvas.add(img);
    });
```
create use two canvas tag one of the canvas have the class 'lower-canvas' and the other (an empty one) has the class :'upper-canvas '. I have to manually delete the canvas everytime I had an image.
`
<canvas id="c" class="lower-canvas" width="600" height="300" style="position: absolute; width: 300px; height: 150px; left: 0px; top: 0px; touch-action: none; -webkit-user-select: none;"></canvas>

<canvas class="upper-canvas " width="600" height="300" style="position: absolute; width: 300px; height: 150px; left: 0px; top: 0px; touch-action: none; -webkit-user-select: none; cursor: default;"></canvas> `
The second canvas is not used.
 Is it a fabric's bug or its there any way to avoid the creation of the second canvas. Thanks for help

**Comment by ShaMan123** [comment, marked as answer]
You should really read the docs...
Use StaticCanvas if you don't want that optimization and if you plan only to draw static images w/o interaction.

**Comment by jeezs** [comment]
Thanks for the answer I’ll read the doc. but I may keep the second Canvas t asI need interactions.
